src/main/java/ch/njol/skript/expressions/ExprSpawn.java

```diff
@@ -38,75 +31,83 @@
+		Skript.registerExpression(ExprSpawn.class, Location.class, ExpressionType.PROPERTY,
+			"[the] spawn[s] [(point|location)[s]] [of %worlds%]",
+			"%worlds%'[s] spawn[s] [(point|location)[s]]"
+		);
 	}
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr((Expression<? extends World>) exprs[0]);
 		return true;
 	}
-	
+
 	@Override
-	protected Location[] get(final Event e, final World[] source) {
-		if (getTime() == -1 && e instanceof SpawnChangeEvent && !Delay.isDelayed(e)) {
+	protected Location[] get(Event e, World[] source) {
+		if (getTime() == -1 && e instanceof SpawnChangeEvent && !Delay.isDelayed(e))
 			return new Location[] {((SpawnChangeEvent) e).getPreviousLocation()};
-		}
-		return get(source, new Converter<World, Location>() {
-			@Override
-			@Nullable
-			public Location convert(final World w) {
-				return w.getSpawnLocation();
-			}
-		});
+		return get(source, World::getSpawnLocation);
 	}
-	
-	@Override
-	public Class<? extends Location> getReturnType() {
-		return Location.class;
-	}
-	
-	@Override
-	public String toString(final @Nullable Event e, final boolean debug) {
-		return "spawn of " + getExpr().toString(e, debug);
-	}
-	
+
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
 		if (mode == ChangeMode.SET)
 			return CollectionUtils.array(Location.class);
 		return null;
 	}
-	
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		assert mode == ChangeMode.SET;
-		assert delta != null;
-		
-		final Location l = (Location) delta[0];
-		final int x = l.getBlockX(), y = l.getBlockY(), z = l.getBlockZ();
-		for (final World w : getExpr().getArray(e)) {
-			w.setSpawnLocation(x, y, z);
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		//noinspection ConstantConditions
+		if (delta == null)
+			return;
+
+		Location originalLocation = (Location) delta[0];
+		assert originalLocation != null;
+		for (World world : getExpr().getArray(e)) {
+			Location location = originalLocation.clone();
+			if (location.getWorld() == null) {
+				location.setWorld(world);
+				world.setSpawnLocation(location);
+			} else if (location.getWorld().equals(world)) {
+				world.setSpawnLocation(location);
+			}
 		}
 	}
-	
-	@SuppressWarnings("unchecked")
+
 	@Override
-	public boolean setTime(final int time) {
+	@SuppressWarnings("unchecked")
+	public boolean setTime(int time) {
 		return super.setTime(time, getExpr(), SpawnChangeEvent.class);
 	}
-	
+
+	@Override
+	public Class<? extends Location> getReturnType() {
```

Comment:
getReturnType and toString go before accept change and change

Also setTime goes before them as well, but there currently is no order, maybe see where other setTime methods order is and then we can add that to code standards.

Reply:
They are in the correct position, see https://github.com/SkriptLang/Skript/blob/master/code-conventions.md#formatting

As for setTime, it pretty much varies across implementing classes as they don't follow the suggested order. The one class I saw that was following the suggested order put it where I did, so it's probably fine.